test/versioned/mysql2/promises.tap.js

```diff
@@ -101,21 +101,19 @@ tap.test('mysql2 promises', { timeout: 30000 }, (t) => {
   t.test('query with options object rather than sql', (t) => {
     return helper
       .runInTransaction(agent, () => {
-        return client.query({ sql: 'SELECT 1' })
-      })
-      .then(() => {
-        agent.getTransaction().end()
+        return client.query({ sql: 'SELECT 1' }).then(() => {
+          agent.getTransaction().end()
+        })
```

Comment:
These tests were weird, I still don't understand how they ever worked, but the new versions make a lot more sense.

Could now omit a call to `agent.getTransaction()` and just pass it in as the first arg to the callback for `runInTransaction`, slightly cleaner.

Reply:
We tend to do the agent.getTransaction() to ensure the current transaction is still the one we're running in. but for this case it's just ending the transaction. I'm going to update but also add an assertion that the transaction we're running in is still the right one